includes/components/class-attribute.php

```diff
@@ -1965,4 +1985,133 @@ public function disable_jetpack_search( $enabled, $query ) {
 
 		return $enabled;
 	}
+
+	/**
+	 * Syncs attributes between models.
+	 *
+	 * @param int           $model_id Model ID.
+	 * @param object|string $model Model object.
+	 */
+	public function sync_model_attributes( $model_id, $model ) {
+
+		// Get sync model name.
+		$sync_model_name = hp\get_array_value( $this->models[ $model::_get_meta( 'name' ) ], 'sync', '' );
+
+		if ( ! $sync_model_name ) {
+			return;
+		}
+
+		// Get sync model.
+		$sync_model = hp\create_class_instance( '\HivePress\Models\\' . $sync_model_name );
+
+		// Check sync model.
+		if ( ! $sync_model ) {
+			return;
+		}
+
+		// Is user sync.
+		$is_user_sync = 'user' === $model::_get_meta( 'name' ) && 'vendor' === $sync_model_name;
+
+		// Get attributes.
+		$attributes      = [];
+		$sync_attributes = [];
+
+		if ( $is_user_sync ) {
+			$attributes = array_filter(
+				hivepress()->attribute->get_attributes( 'user' ),
+				function( $attribute ) {
+					return hp\get_array_value( $attribute, 'synced' );
+				}
+			);
+
+			$sync_attributes = hivepress()->attribute->get_attributes( 'vendor' );
+		} else {
+			$attributes = array_filter(
+				hivepress()->attribute->get_attributes( $sync_model_name ),
+				function( $attribute ) {
+					return hp\get_array_value( $attribute, 'synced' );
+				}
+			);
+
+			$sync_attributes = hivepress()->attribute->get_attributes( $model::_get_meta( 'name' ) );
+		}
+
+		// Check attributes.
+		if ( ! $attributes ) {
+			return;
+		}
+
+		// Get values.
+		$values = array_intersect_key( $model->serialize(), $attributes );
+
+		foreach ( $attributes as $attribute_name => $attribute ) {
+			if ( ! isset( $attribute['edit_field']['options'] ) || isset( $attribute['edit_field']['_external'] ) ) {
+				continue;
+			}
+
+			// Get field.
+			$attribute_field = hp\get_array_value( $model->_get_fields(), $attribute_name );
+
+			if ( ! $attribute_field || ! $attribute_field->get_value() || ! isset( $sync_attributes[ $attribute_name ]['edit_field']['option_args']['taxonomy'] ) ) {
+				continue;
+			}
+
+			// Get taxonomy.
+			$term_names_taxonomy = $attribute['edit_field']['option_args']['taxonomy'];
+			$term_ids_taxonomy   = $sync_attributes[ $attribute_name ]['edit_field']['option_args']['taxonomy'];
+
+			if ( ! $is_user_sync ) {
+				$term_names_taxonomy = $sync_attributes[ $attribute_name ]['edit_field']['option_args']['taxonomy'];
+				$term_ids_taxonomy   = $attribute['edit_field']['option_args']['taxonomy'];
+			}
+
+			// Get term names.
+			$term_names = get_terms(
+				[
+					'taxonomy'   => $term_names_taxonomy,
+					'include'    => (array) $attribute_field->get_value(),
+					'fields'     => 'names',
+					'hide_empty' => false,
+				]
+			);
+
+			if ( ! $term_names ) {
+				continue;
+			}
+
+			// Get term IDs.
+			$term_ids = get_terms(
```

Comment:
Let me know if you resolved syncing of selectable attributes, if not then this is probably due to some difference in how user's selectable attributes are implemented, because the code below (moved from the Vendor component) correctly matched term IDs to sync selectable attributes, but both listing and vendor are post-based models unlike users.

Reply:
Yes, these types of fields were fixed